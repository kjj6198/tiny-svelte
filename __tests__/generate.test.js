const generate = require("../generate");

describe("Genretate", () => {
  it("should generate code (pure element)", () => {
    const code = generate("<div><p></p></div>");
    expect(code).toMatchSnapshot();
  });

  it("should generate code (element with text)", () => {
    const code = generate("<div><p>345</p><p>12345</p></div>");
    expect(code).toMatchSnapshot();
  });

  it("should generate code attr", () => {
    const code = generate('<div><p role="status">345</p><p>12345</p></div>');
    expect(code).toMatchSnapshot();
  });
});
