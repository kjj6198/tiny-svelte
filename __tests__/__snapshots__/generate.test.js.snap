// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Genretate should generate code (element with text) 1`] = `
"
    function append(node, target) {
      target.appendChild(node);
    }

    let fragment = document.createDocumentFragment();
  let div1 = document.createElement('div');
let p1 = document.createElement('p');
let t1 = document.createTextNode(\\"345\\");

        append(
          t1,
          p1
        );

        append(
          p1,
          div1
        );
let p2 = document.createElement('p');
let t2 = document.createTextNode(\\"12345\\");

        append(
          t2,
          p2
        );

        append(
          p2,
          div1
        );

        append(
          div1,
          fragment
        );
document.body.appendChild(fragment);"
`;

exports[`Genretate should generate code (pure element) 1`] = `
"
    function append(node, target) {
      target.appendChild(node);
    }

    let fragment = document.createDocumentFragment();
  let div1 = document.createElement('div');
let p1 = document.createElement('p');

        append(
          p1,
          div1
        );

        append(
          div1,
          fragment
        );
document.body.appendChild(fragment);"
`;

exports[`Genretate should generate code attr 1`] = `
"
    function append(node, target) {
      target.appendChild(node);
    }

    let fragment = document.createDocumentFragment();
  let div1 = document.createElement('div');
let p1 = document.createElement('p');
p1.setAttribute(role, status)let t1 = document.createTextNode(\\"345\\");

        append(
          t1,
          p1
        );

        append(
          p1,
          div1
        );
let p2 = document.createElement('p');
let t2 = document.createTextNode(\\"12345\\");

        append(
          t2,
          p2
        );

        append(
          p2,
          div1
        );

        append(
          div1,
          fragment
        );
document.body.appendChild(fragment);"
`;
